### PR TITLE
refactor(theme): change legacy support default value to false

### LINKED ISSUE #6803: THEME: Use CSS variables by default
Switch to using CSS variables by default, i.e. legacy support should be set to false by default.

Items:

- [x] Update README.md (if applicable)
- [x] Update DocFX documentation
- [x] Update Sassdoc

### PR DESCRIPTION
**Pull request by didimmova**
Closes #6803

### Additional information (check all that apply):
 - [ ] Bug fix
 - [ ] New functionality
 - [x] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [x] All relevant tags have been applied to this PR
 - [ ] This PR includes unit tests covering all the new code
 - [ ] This PR includes API docs for newly added methods/properties
 - [ ] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [x] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [x] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

**Comment by didimmova** [comment]
@simeonoff They already are?

File: projects/igniteui-angular/src/lib/core/styles/themes/_index.scss [hunk 1]
```diff
@@ -51,7 +51,7 @@
 /// @param {Map} $palette - An igx-palette to be used by the global theme.
 /// @param {Map} $schema [$light-schema] - The schema used as basis for styling the components.
 /// @param {List} $exclude [( )] - A list of igx components to be excluded from the global theme styles.
-/// @param {Boolean} $legacy-support - Turn off support for IE11, allowing you to use css variables to style components.
+/// @param {Boolean} $legacy-support [false] - If set to false, it turns off support for IE11, allowing you to use css variables to style components.
```
Comment (simeonoff) [inline projects/igniteui-angular/src/lib/core/styles/themes/_index.scss]:
```suggestion
/// @param {Boolean} $legacy-support [false] - If set to true, it turns on support for IE11, i.e. css variables are not used in the generated component themes.
```

File: projects/igniteui-angular/src/lib/core/styles/themes/_index.scss [hunk 2]
```diff
@@ -412,11 +412,11 @@
 /// Creates a global theme that can be used with light backgrounds.
 /// @param {Map} $palette - An igx-palette to be used by the global theme.
 /// @param {List} $exclude [( )] - A list of igx components to be excluded from the global theme styles.
-/// @param {Boolean} $legacy-support - Turn off support for IE11, allowing you to use css variables to style components.
+/// @param {Boolean} $legacy-support [false] - If set to false, it turns off support for IE11, allowing you to use css variables to style components.
```
Comment (simeonoff) [inline projects/igniteui-angular/src/lib/core/styles/themes/_index.scss]:
```suggestion
/// @param {Boolean} $legacy-support [false] - If set to true, it turns on support for IE11, i.e. css variables are not used in the generated component themes.
```

File: projects/igniteui-angular/src/lib/core/styles/themes/_index.scss [hunk 3]
```diff
@@ -447,11 +447,11 @@
 /// Creates a global theme that can be used with dark backgrounds.
 /// @param {Map} $palette - An igx-palette to be used by the global theme.
 /// @param {List} $exclude [( )] - A list of igx components to be excluded from the global theme styles.
-/// @param {Boolean} $legacy-support - Turn off support for IE11, allowing you to use css variables to style components.
+/// @param {Boolean} $legacy-support [false] - If set to false, it turns off support for IE11, allowing you to use css variables to style components.
```
Comment (simeonoff) [inline projects/igniteui-angular/src/lib/core/styles/themes/_index.scss]:
```suggestion
/// @param {Boolean} $legacy-support [false] - If set to true, it turns on support for IE11, i.e. css variables are not used in the generated component themes.
```

File: projects/igniteui-angular/src/lib/core/styles/themes/_index.scss [hunk 4]
```diff
@@ -482,11 +482,11 @@
 /// Creates a global fluent theme that can be used with light backgrounds.
 /// @param {Map} $palette - An igx-palette to be used by the global theme.
 /// @param {List} $exclude [( )] - A list of igx components to be excluded from the global theme styles.
-/// @param {Boolean} $legacy-support - Turn off support for IE11, allowing you to use css variables to style components.
+/// @param {Boolean} $legacy-support [false] - If set to false, it turns off support for IE11, allowing you to use css variables to style components.
```
Comment (simeonoff) [inline projects/igniteui-angular/src/lib/core/styles/themes/_index.scss]:
```suggestion
/// @param {Boolean} $legacy-support [false] - If set to true, it turns on support for IE11, i.e. css variables are not used in the generated component themes.
```

File: projects/igniteui-angular/src/lib/core/styles/themes/_index.scss [hunk 5]
```diff
@@ -517,11 +517,11 @@
 /// Creates a global fluent theme that can be used with dark backgrounds.
 /// @param {Map} $palette - An igx-palette to be used by the global theme.
 /// @param {List} $exclude [( )] - A list of igx components to be excluded from the global theme styles.
-/// @param {Boolean} $legacy-support - Turn off support for IE11, allowing you to use css variables to style components.
+/// @param {Boolean} $legacy-support [false] - If set to false, it turns off support for IE11, allowing you to use css variables to style components.
```
Comment (simeonoff) [inline projects/igniteui-angular/src/lib/core/styles/themes/_index.scss]:
```suggestion
/// @param {Boolean} $legacy-support [false] - If set to true, it turns on support for IE11, i.e. css variables are not used in the generated component themes.
```

File: projects/igniteui-angular/src/lib/core/styles/themes/_index.scss [hunk 6]
```diff
@@ -553,11 +553,11 @@
 /// Creates a global bootstrap-like theme that can be used with light backgrounds.
 /// @param {Map} $palette - An igx-palette to be used by the global theme.
 /// @param {List} $exclude [( )] - A list of igx components to be excluded from the global theme styles.
-/// @param {Boolean} $legacy-support - Turn off support for IE11, allowing you to use css variables to style components.
+/// @param {Boolean} $legacy-support [false] - If set to false, it turns off support for IE11, allowing you to use css variables to style components.
```
Comment (simeonoff) [inline projects/igniteui-angular/src/lib/core/styles/themes/_index.scss]:
```suggestion
/// @param {Boolean} $legacy-support [false] - If set to true, it turns on support for IE11, i.e. css variables are not used in the generated component themes.
```

File: projects/igniteui-angular/src/lib/core/styles/themes/_index.scss [hunk 7]
```diff
@@ -615,11 +615,11 @@
 /// Creates a global bootstrap-like theme that can be used with dark backgrounds.
 /// @param {Map} $palette - An igx-palette to be used by the global theme.
 /// @param {List} $exclude [( )] - A list of igx components to be excluded from the global theme styles.
-/// @param {Boolean} $legacy-support - Turn off support for IE11, allowing you to use css variables to style components.
+/// @param {Boolean} $legacy-support [false] - If set to false, it turns off support for IE11, allowing you to use css variables to style components.
```
Comment (simeonoff) [inline projects/igniteui-angular/src/lib/core/styles/themes/_index.scss]:
```suggestion
/// @param {Boolean} $legacy-support [false] - If set to true, it turns on support for IE11, i.e. css variables are not used in the generated component themes.
```

**Pull request merged**